### PR TITLE
Add check to stop sending invite to participant team if user's email is in blocked email domains

### PR DESCRIPTION
**Pull request by Sanji515**
#### Changes proposed in this pull request:

- Added checks in `challenges/utils.py` for `blocked email domains` and `allowed email domains`.
- Updated `add_participant_to_team` with these common checks.
- Added checks in `invite_participant_to_team` api with these checks.
- Added unit tests for the added checks in `participants/test_views.py`

### CODE REVIEW

File: apps/challenges/utils.py [hunk 1]
```diff
@@ -189,3 +189,20 @@ def create_federated_user(name, repository, aws_keys):
         DurationSeconds=43200,
     )
     return response
+
+
+def check_if_user_is_in_allowed_email_domains(email, challenge_pk):
```
Comment (RishabhJain2018) [inline apps/challenges/utils.py]: `check_if_user_is_in_allowed_email_domains` -> `is_user_in_allowed_email_domains`

File: apps/challenges/utils.py [hunk 2]
```diff
@@ -189,3 +189,20 @@ def create_federated_user(name, repository, aws_keys):
...
+    challenge = get_challenge_model(challenge_pk)
+    for domain in challenge.allowed_email_domains:
+        if domain.lower() in email.lower():
+            return True
+    return False
+
+
+def check_if_user_is_in_blocked_email_domains(email, challenge_pk):
```
Comment (RishabhJain2018) [inline apps/challenges/utils.py]: `check_if_user_is_in_blocked_email_domains` -> `is_user_in_blocked_email_domains`

File: apps/challenges/views.py [hunk 3]
```diff
@@ -52,6 +52,8 @@
     get_challenge_phase_split_model,
     get_dataset_split_model,
     get_leaderboard_model,
+    check_if_user_is_in_allowed_email_domains,
```
Comment (RishabhJain2018) [inline apps/challenges/views.py]: Please replace them from above.

File: apps/participants/views.py [hunk 4]
```diff
@@ -181,6 +185,34 @@ def invite_participant_to_team(request, pk):
         [participant_team]
     ).values_list("id", flat=True)
 
+    if len(team_participated_challenges) > 0:
+        for challenge_pk in team_participated_challenges:
+            challenge = Challenge.objects.get(pk=challenge_pk)
```
Comment (RishabhJain2018) [inline apps/participants/views.py]: Please use `get_challenge_model`

**Comment by Sanji515** [comment]
> @Sanji515 I think we should also add banned email ID's check: https://github.com/Cloud-CV/EvalAI/pull/2493/files#diff-ff54996dc8925e4c7dbde340701c7086L260 in this API: https://github.com/Cloud-CV/EvalAI/pull/2493/files#diff-5723745522a724a151ce8c3465afe0f1R189

yes yes, Done @rishabhbhardwaj Please review.

**Pull request merged**